File: graf.cpp
#include <iostream>
#include <vector>

using namespace std;

class Graph {
private:
    int n;  // количество вершин
    vector<vector<int>> adjMatrix;  // матрица смежности

public:
    Graph(int vertices) {
        n = vertices;
        adjMatrix.resize(n, vector<int>(n, 0));
    }

    void addEdge(int u, int v) {
        adjMatrix[u][v] = 1;  // ребро из u в v
        adjMatrix[v][u] = 1;  // для неориентированного графа
    }

    void printGraph() {
        for (int i = 0; i < n; i++) {
            cout << "Вершина " << i << ": ";
            for (int j = 0; j < n; j++) {
                cout << adjMatrix[i][j] << " ";
            }
            cout << endl;
        }
    }
};

int main() {
    Graph g(5);  // создаем граф с 5 вершинами

    g.addEdge(0, 1);
    g.addEdge(0, 4);
    g.addEdge(1, 2);
    g.addEdge(1, 3);
    g.addEdge(1, 4);
    g.addEdge(2, 3);
    g.addEdge(3, 4);

    g.printGraph();

    return 0;
}

File: graf.java
import java.util.ArrayList;
import java.util.List;

public class Graph {
    private List<List<Integer>> adjList;

    public Graph(int vertices) {
        adjList = new ArrayList<>();
        for (int i = 0; i < vertices; i++) {
            adjList.add(new ArrayList<>());
        }
    }

    // Метод добавления ребра между вершинами u и v (неориентированный граф)
    public void addEdge(int u, int v) {
        adjList.get(u).add(v);
        adjList.get(v).add(u);
    }

    // Получение списка смежных вершин для данной вершины
    public List<Integer> getAdjVertices(int vertex) {
        return adjList.get(vertex);
    }

    public static void main(String[] args) {
        Graph graph = new Graph(5); // Создаем граф с 5 вершинами

        // Добавляем ребра
        graph.addEdge(0, 1);
        graph.addEdge(0, 4);
        graph.addEdge(1, 2);
        graph.addEdge(1, 3);
        graph.addEdge(1, 4);
        graph.addEdge(2, 3);
        graph.addEdge(3, 4);

        // Выводим список смежных вершин для вершины 1
        System.out.println("Смежные вершины для вершины 1: " + graph.getAdjVertices(1));
    }
}
